sdk/webpubsub/azure-webpubsub-client/azure/webpubsub/client/_model_base.py

```diff
@@ -0,0 +1,706 @@
+# license information.
+# --------------------------------------------------------------------------
+# pylint: disable=protected-access, arguments-differ, signature-differs, broad-except
+# pyright: reportGeneralTypeIssues=false
+
+import functools
+import sys
+import logging
+import base64
+import re
+import copy
+import typing
+from datetime import datetime, date, time, timedelta, timezone
+from json import JSONEncoder
+import isodate
+from azure.core.exceptions import DeserializationError
+from azure.core import CaseInsensitiveEnumMeta
+from azure.core.pipeline import PipelineResponse
+from azure.core.serialization import NULL as AzureCoreNull
+
+if sys.version_info >= (3, 9):
+    from collections.abc import MutableMapping
+else:
+    from typing import MutableMapping
+
+_LOGGER = logging.getLogger(__name__)
+
+__all__ = ["NULL", "AzureJSONEncoder", "Model", "rest_field", "rest_discriminator"]
+
+
+class _Null(object):
+    """To create a Falsy object"""
+
+    def __bool__(self):
+        return False
+
+    __nonzero__ = __bool__  # Python2 compatibility
+
+
+NULL = _Null()
+"""
+A falsy sentinel object which is supposed to be used to specify attributes
+with no data. This gets serialized to `null` on the wire.
+"""
+
+TZ_UTC = timezone.utc
+
+
+def _timedelta_as_isostr(td: timedelta) -> str:
+    """Converts a datetime.timedelta object into an ISO 8601 formatted string, e.g. 'P4DT12H30M05S'
+
+    Function adapted from the Tin Can Python project: https://github.com/RusticiSoftware/TinCanPython
+
+    :param timedelta td: The timedelta to convert
+    :rtype: str
+    :return: ISO8601 version of this timedelta
+    """
+
+    # Split seconds to larger units
+    seconds = td.total_seconds()
+    minutes, seconds = divmod(seconds, 60)
+    hours, minutes = divmod(minutes, 60)
+    days, hours = divmod(hours, 24)
+
+    days, hours, minutes = list(map(int, (days, hours, minutes)))
+    seconds = round(seconds, 6)
+
+    # Build date
+    date_str = ""
+    if days:
+        date_str = "%sD" % days
+
+    # Build time
+    time_str = "T"
+
+    # Hours
+    bigger_exists = date_str or hours
+    if bigger_exists:
+        time_str += "{:02}H".format(hours)
+
+    # Minutes
+    bigger_exists = bigger_exists or minutes
+    if bigger_exists:
+        time_str += "{:02}M".format(minutes)
+
+    # Seconds
+    try:
+        if seconds.is_integer():
+            seconds_string = "{:02}".format(int(seconds))
+        else:
+            # 9 chars long w/ leading 0, 6 digits after decimal
+            seconds_string = "%09.6f" % seconds
+            # Remove trailing zeros
+            seconds_string = seconds_string.rstrip("0")
+    except AttributeError:  # int.is_integer() raises
+        seconds_string = "{:02}".format(seconds)
+
+    time_str += "{}S".format(seconds_string)
+
+    return "P" + date_str + time_str
+
+
+def _datetime_as_isostr(dt: typing.Union[datetime, date, time, timedelta]) -> str:
```

Comment:
Is this equivalent to [`_convert_to_isoformat`](https://github.com/Azure/azure-sdk-for-python/blob/faa1d1a32912e1c508948bc613508fab283e367a/sdk/core/azure-core/azure/core/utils/_utils.py#L48) in azure.core.utils ? There might be other functions in there that can be used too

Reply:
The .py is same with https://github.com/Azure/autorest.python/blob/main/packages/autorest.python/autorest/codegen/templates/model_base.py.jinja2 which will be moved into `azure.core` finally. So I think we can just stay same and optimize it gradually in the future.